src/solve.jl

```diff
@@ -533,10 +533,12 @@ function DiffEqBase.__init(
 end
 
 function DiffEqBase.__init(prob::ODEProblem, ::Nothing, args...; kwargs...)
-    DiffEqBase.__init(prob, DefaultODEAlgorithm(), args...; kwargs...)
+    alg = DiffEqBase.prepare_alg(DefaultODEAlgorithm(), prob.u0, prob.p, prob)
```

Comment:
Why prepare the alg here? That will happen inside the DiffEqBase pipeline.

Reply:
form what I can see of other packages that provide default algorithms, this is the standard way of doing so. This is pretty much copied from both `NonlinearSolve.jl` and `DifferentialEquations.jl`

Reply:
No, it doesn't match this https://github.com/SciML/DifferentialEquations.jl/blob/master/src/default_solve.jl#L14